### DISCUSSION TITLE
Do not merge "master" into branches

### DISCUSSION THREAD

**Discussion by easybe** [UpstreamDataInc/goosebit#169] [General]
We should consider disabling the "Always suggest updating pull request branches" setting as the GitHub feature will merge "master" into the feature branch, which should be avoided. The branch should be rebased onto "master" instead. Not sure if the behavior of the button can be changed.

**Comment by jameshilliard** [comment]
I disabled everything except rebase merging just now, let me know if that looks better.

**Reply by easybe** [reply]
Merging feature branches into "master" is fine. The other direction is the problem. I would stick to using merge commits for PRs, as it requires less discipline from the the contributors to create nice commits.

**Reply by easybe** [reply]
Disabling this one should be enough:
![image](https://github.com/user-attachments/assets/b2e74cdd-763f-4f4e-bb68-b4b1bd0922ed)

**Comment by easybe** [comment]
So, is it decided that the merge strategy is "rebase only" from now on? Not what I was aiming for, but works for me.
I guess. alternatively we would just need to be disciplined enough when "updating" branches through the user interface: 
![image](https://github.com/user-attachments/assets/78bbfc29-32b2-42d1-a6df-deba7b4bfd72)
Merges should really only go into "master" and not the other way.

**Reply by b-rowan** [reply]
I'll see if I can set the settings up so that is the only option.

**Reply by b-rowan** [reply]
See https://github.com/orgs/community/discussions/12032

I have tried to do this with the "Require linear history" ruleset, but I have no idea if that will work.

**Comment by easybe** [comment]
As we now integrate PRs with the “rebase only” strategy, this is no longer an issue.
